src/frontend/e2e/tests/upload.spec.ts

```diff
@@ -1,65 +1,92 @@
 import { test, expect } from "@playwright/test";
-import { UploadSample } from "../pages/upload";
+import { uploadSampleFiles } from "../utils/upload";
+import { createSampleUploadData } from "../utils/sample";
 
+const locations = [
+  "Africa/Angola/Luanda",
+  "Europe/Russia/Kaluga",
+  "Asia/China",
+];
 test.describe("Upload sample tests", () => {
   const dateErrorMessage = "Update format to YYYY-MM-DD";
-  const fileExtensions = [".txt", ".fasta", ".fa"]; //todo zip and gzip
+  const fileExtensions = [".txt", ".fa", ".fasta"]; //todo: add zip and gzip
   test.beforeEach(async ({ page }, workerInfo) => {
     const baseUrl = workerInfo.config.projects[0].use.baseURL;
     const url = `${baseUrl}/data/samples`;
     await page.goto(url);
+    //accept cookie t&c (if prompted and not in CI)
+    const tAndCSelector =
+      '[aria-label="Help us improve CZ GEN EPI"] >> text=Accept';
+    const tAndC = page.locator(tAndCSelector);
+    if (await tAndC.isVisible()) {
+      await page.locator(tAndCSelector).click();
+    }
+    //click upload button
+    await page.locator('[data-test-id="upload-btn"]').click();
```

Comment:
why doesn't this make use of basePage helpers?

Reply:
This has been out there before page object but I have now migrated.